src/parse-nm.c

```diff
@@ -55,6 +55,14 @@ type_from_str(const char* type_str)
     else if (!g_strcmp0(type_str, "ip-tunnel") || !g_strcmp0(type_str, "wireguard"))
         return NETPLAN_DEF_TYPE_TUNNEL;
     */
+    /* TODO: Vxlans are not yet fully supported by the keyfile parser
+    else if (!g_strcmp0(type_str, "vxlan"))
+        return NETPLAN_DEF_TYPE_VXLAN;
+    */
+    /* TODO: Vrfs are not yet fully supported by the keyfile parser
+    else if (!g_strcmp0(type_str, "vrf"))
+        return NETPLAN_DEF_TYPE_VRF;
+    */
```

Comment:
I'm pretty sure this is supported in the keyfile format these days... @thom311, @lkundrak, @tyll, do any of you know?

Reply:
NetworkManager's keyfile format supports (necessarily) everything that NetworkManager supports.

But I think the reference to "the keyfile parser" here is `src/parse-nm.c` from netplan itself. This branch doesn't seem to implement the NetworkManager backend (which is a fine choice).

For the record, NetworkManager supports vxlan and vrfs.

Reply:
That's correct - I intentionally did not touch the nm parsers in this pr. Thanks!

Reply:
@thom311 Is there any documentation of this stuff on networkmanager.dev? I couldn't find it specifically.

@netdever While I'm not part of the project, I would personally prefer to see things like this being implemented for both backends.

Reply:
in NM, you always create "connection profiles". So, to create a vxlan/vrf configuration, you also create appropriate profiles.  The fields of a profile are documented in `man nm-settings-{dbus,nmcli,keyfile}` (which is also on the website). But there is no direct documentation how to create a vxlan/vrf configuration.

Reply:
FTR: NetworkManager's VRF and VXLAN settings are listed here: https://networkmanager.dev/docs/api/latest/nm-settings-nmcli.html#id-1.2.9.4.43

But as Thomas stated, this code snippet in `src/parse-nm.c` is from netplan itself (allowing it to read a NM keyfile connection profile) and not necessarily needed for this PR.